README.md

```diff
@@ -1 +1,277 @@
+ * `options.onProgress` (function, optional) - callback that is called after finishing execution of each migration to report progress.
+ * `options.isDryRun` (bool, optional) - flag that indicates if it is a dry run that should give the same output as running a migration but without making any actual changes.
+ 
+### `getLatestMigrationVersion() -> int`
+
+Return the version of the latest migration.
+
+## CLI
+
+The CLI is a NodeJS binary named `jsipfs-repo-migrations`. 
+It has several commands:
+
+ * `migrate` - performs forward/backward migration to specific or latest version.
+ * `status` - check repo for migrations that should be run.
+ * `add` - bootstraps new migration.
+ 
+For further details see the `--help` pages.
+
+## Creating a new migration
+
+Migrations are one of those things that can be extremely painful on users. At the end of the day, we want users never to have to think about it. The process should be:
+
+- SAFE. No data lost. Ever.
+- Revertible. Tools must implement forward and backward (if possible) migrations.
+- Tests. Migrations have to be well tested.
+- To Spec. The tools must conform to the spec.
+
+If your migration has several parts, it should be fail-proof enough that if one part of migration fails the previous changes
+are reverted before propagating the error. If possible then the outcome should be consistent repo so it migration could
+be run again. 
+
+### Architecture of a migration
+
+All migrations are placed in the `/migrations` folder. Each folder there represents one migration that follows the migration
+API.
+
+All migrations are collected in `/migrations/index.js`, which should not be edited manually. It is regenerated on
+every run of `jsipfs-migrations add` (manual changes should follow the same style of modifications). 
+**The order of migrations is important and migrations must be sorted in ascending order**.
+
+Each migration must follow this API. It must export an object in its `index.js` that has following properties:
+
+ * `version` (int) - Number that represents the version which the repo will migrate to (eg. `migration-8` will move the repo to version 8).
+ * `description` (string) - Brief description of what the migrations does.
+ * `migrate` (function) - Function that performs the migration (see signature of this function below)
+ * `revert` (function) - If defined then this function will revert the migration to the previous version. Otherwise it is assumed that it is not possible to revert this migration.
+
+#### `.migrate(repoPath, isBrowser)`
+
+_Do not confuse this function with the `require('ipfs-repo-migrations').migrate()` function that drives the whole migration process!_
+
+Arguments:
+ * `repoPath` (string) - absolute path to the root of the repo
+ * `options` (object, optional) - object containing `IPFSRepo` options, that should be used to construct a datastore instance.
+ * `isBrowser` (bool) - indicates if the migration is run in a browser environment (as opposed to NodeJS)
```

Comment:
You mean as part of `options`/future `repoOptions`? If so, then I don't think so, because these options should be linked only to constructions of datastore and should follow the same semantic as the object passed to `IPFSRepo`...